File: app/core/entities/task.rb
# frozen_string_literal: true

module Entities
  # Task entity
  class Task
    attr_accessor :id, :title, :description, :user_id, :status

    def initialize(attributes = {})
      @id = attributes[:id]
      @title = attributes[:title]
      @description = attributes[:description]
      @user_id = attributes[:user_id]
      @status = attributes[:status]
    end
  end
end

File: client-app/src/application/services/transformers/task.ts
import { ITask, ServerTask, TaskResponse } from "../../../domain/models/Task";

export const transformApiPostToDomain = (apiPost: ServerTask): ITask => {
  return {
    id: apiPost.id.toString(),
    title: apiPost.attributes.title,
    description: apiPost.attributes.description,
    status: apiPost.attributes.status,
    status_human_readable: apiPost.attributes.status_human_readable,
  } as unknown as ITask; // FIXME: this typecast
};

export const transformApiPostsToDomain = (apiPosts: TaskResponse): ITask[] => {
  return apiPosts.data.map(transformApiPostToDomain);
};
